### PR TITLE
Automatically animated UI icons 

### PR DESCRIPTION
**Pull request by Macoron**
### Purpose
Previously each animated item should had custom script to update animation frames in UI slots. That lead to a lot stupid, inefficient and repeatable code in different scripts with same line `pickupable.RefreshUISlotImage();`
This PR implements automatic way for UI_ItemImage to update image state based on current SpriteHandler state. Now SpriteHandler fires event when animation frame changed. Subscribed UI_ItemImage change its sprite respectively.

### Notes:
UI image will not be updated if SpriteRender sprite was replaced or animated by another script. It only works if SpriteHandlers have full control over SpriteRender. 
Not all occurrence of `pickupable?.RefreshUISlotImage();` was replaced or deleted. Old approach is still working, but it should be replaced in the future.
Cigarette, lighters and stun button was refactored for new approach. Other items should be refactored in different PR.
Energy sword was broken before this PR. Though now inventory animation works correctly, clothing animation still funny.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Sprite Handler/SpriteHandler.cs [hunk 1]
```diff
@@ -423,8 +482,9 @@ private void SetImageSprite(Sprite value)
 			{
 				image.enabled = false;
 			}
-
 		}
+
+		OnSpriteChanged?.Invoke(value);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Sprite Handler/SpriteHandler.cs:487]: how much of a performance impact is this?

**Comment by Bod9001** [comment]
This technically covers
#2593
#2592
you want to take the Bounties?

File: UnityProject/Assets/Scripts/Sprite Handler/SpriteHandler.cs (hunk 1 above)
Comment (Macoron) [reply UnityProject/Assets/Scripts/Sprite Handler/SpriteHandler.cs:487]: This event invokes for each frame in object animation (gap between frames usually 0.2-0.4 second). For non-animated items or objects it is just one null check. It's nothing.
For animated items this event just replace sprite in UI Image class. Performance here is even better than before. In old approach custom script used to create coroutine or even Update cycle to manually refresh UI slot.

Keep in mind that this is Client side code only. From my test I can't see any impact on performance.

**Comment by Macoron** [comment]
Tbh, I'm not really interested in bounties and can return it to pool. 
Hope that I didn't break anything in progress, but issues seems to be complete.

**Comment by Bod9001** [comment]
cool, loooks good,merging

**Pull request merged**